osu.Game/Screens/Play/GameplayClockContainer.cs

```diff
@@ -198,6 +194,14 @@ private void updateRate()
         protected override void Dispose(bool isDisposing)
         {
             base.Dispose(isDisposing);
+
+            removeSourceClockAdjustments();
```

Comment:
I'm a little weary that this isn't going to happen in time due to async disposal...

Reply:
It's still called in `StopUsingBeatmapClock`, which is called in `OnExiting`. I just combined them because it makes more sense.

Reply:
Ah I missed that.